packages/editor/src/package/storage-format.ts

```diff
@@ -65,25 +91,47 @@ export function createEmptyDocument(
   }
 }
 
-export function migrate(stateBeforeMigration: unknown): {
+export function migrate(
+  stateBeforeMigration: unknown,
+  variant: EditorVariant
+): {
   migratedState: StorageFormat
   stateChanged: boolean
 } {
-  if (!t.type({ version: t.number }).is(stateBeforeMigration))
+  let migratedState: StorageFormat
+  let stateChanged = false
+
+  // Check if the state is the (old format)
+  if (
+    isValidDocument(stateBeforeMigration) &&
+    !StorageFormatType.is(stateBeforeMigration)
+  ) {
+    migratedState = {
+      id: uuid_v4(),
+      type: documentType,
+      variant,
+      domainOrigin: window.location.origin,
+      version: 0,
+      editorVersion: getEditorVersion(),
+      dateModified: getCurrentDatetime(),
+      document: stateBeforeMigration,
+    }
+    stateChanged = true
+  } else if (StorageFormatType.is(stateBeforeMigration)) {
+    // deep copy
+    migratedState = JSON.parse(
+      JSON.stringify(stateBeforeMigration)
+    ) as StorageFormat
+  } else {
     throw new Error(
-      `Missing property 'version: number' in state while trying to perform migrations. Got ${JSON.stringify(stateBeforeMigration)}`
+      `Unknown, invalid state format. Got ${JSON.stringify(stateBeforeMigration)}`
     )
+  }
```

Comment:
This will throw the error if we get a state that is neither the (old) document state nor the newest StorageFormat state, so all possible states in between.

Reply:
Maybe something like: 
```ts
if (isOldDocumentFormat(stateBeforeMigration)) {
  migratedState = {
    id: uuid_v4(),
    type: documentType,
    variant,
    domainOrigin: window.location.origin,
    version: 0,
    editorVersion: getEditorVersion(),
    dateModified: getCurrentDatetime(),
    document: JSON.parse(JSON.stringify(stateBeforeMigration)),
  }
  stateChanged = true
} else {
  migratedState = JSON.parse(JSON.stringify(stateBeforeMigration))
}
```